plugins/generator-1.18.2/forge-1.18.2/templates/rangeditem/rangeditem.java.ftl

```diff
@@ -143,17 +147,20 @@ public class ${name}Item extends Item {
 			if (stack.hurt(1, world.getRandom(), entity)) {
 				stack.shrink(1);
 				stack.setDamageValue(0);
-            	if (stack.isEmpty())
-               		entity.getInventory().removeItem(stack);
+				if (stack.isEmpty())
+					entity.getInventory().removeItem(stack);
 			}
 		} else{
 			stack.shrink(1);
-            if (stack.isEmpty())
-               entity.getInventory().removeItem(stack);
+			if (stack.isEmpty())
+			   entity.getInventory().removeItem(stack);
 		}

```

Comment:
Add a `<#if !data.pickupProjectiles>entityarrow.pickup = AbstractArrow.Pickup.DISALLOWED;</#if>` to fix the issue with projectiles with an ammo